tool/src/cpp/ty.rs

```diff
@@ -85,6 +85,33 @@ impl<'ccx, 'tcx: 'ccx> TyGenContext<'ccx, 'tcx, '_> {
             .flat_map(|method| self.gen_method_info(id, method))
             .collect::<Vec<_>>();
 
+        let mut found_default: Option<&hir::EnumVariant> = None;
+        let mut found_zero = None;
+
+        // Not all enums have a zero-variant; zero-initializing those is a mistake and will
+        // lead to aborts in the conversion code. To allow default-initialization, we generate *some*
+        // default ctor. It is, in order: the explicit default variant, OR the variant with 0 discriminant,
+        // OR the first variant.
+        for v in ty.variants.iter() {
+            if v.attrs.default {
+                if let Some(existing) = found_default {
+                    self.errors.push_error(format!(
+                        "Found multiple default variants for enum: {} and {}",
+                        existing.name, v.name
+                    ))
+                }
+                found_default = Some(v)
+            }
+            if v.discriminant == 0 {
+                found_zero = Some(v)
+            }
+        }
+
+        let default_variant = found_default
+            .or(found_zero)
+            .unwrap_or(ty.variants.first().unwrap());
```

Comment:
Issue: Please open a follow-up to require `#[default]` (or _maybe_ discriminant 0). Implicitly defaulting to the first value is bad.

Reply:
https://github.com/rust-diplomat/diplomat/issues/885